integration/spark/shared/src/main/java/io/openlineage/spark/agent/facets/IcebergCommitMetrics.java

```diff
@@ -0,0 +1,44 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark.agent.facets;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Builder
+@EqualsAndHashCode
+public class IcebergCommitMetrics {
+  Long totalDuration;
+  Long attempts;
+  Long addedDataFiles;
+  Long removedDataFiles;
+  Long totalDataFiles;
+  Long addedDeleteFiles;
+  Long addedEqualityDeleteFiles;
+  Long addedPositionalDeleteFiles;
+  Long addedDVs;
+  Long removedDeleteFiles;
+  Long removedEqualityDeleteFiles;
+  Long removedPositionalDeleteFiles;
+  Long removedDVs;
+  Long totalDeleteFiles;
+  Long addedRecords;
+  Long removedRecords;
+  Long totalRecords;
+  Long addedFilesSizeInBytes;
+  Long removedFilesSizeInBytes;
+  Long totalFilesSizeInBytes;
+  Long addedPositionalDeletes;
+  Long removedPositionalDeletes;
+  Long totalPositionalDeletes;
+  Long addedEqualityDeletes;
+  Long removedEqualityDeletes;
+  Long totalEqualityDeletes;
```

Comment:
Because we don't want to serialize empty values to `0`.

What we do, is basically expressing Iceberg Metrics classes like this -> https://github.com/apache/iceberg/blob/apache-iceberg-1.7.1/core/src/main/java/org/apache/iceberg/metrics/ScanMetricsResult.java with our facet. Iceberg metrics reports contain dozens of nullable `CounterResult`. In case of null, we don't want to set the value within our class.  Then Jackson will skip serialisation of those fields and serialize only the non-null metrics from icberg report. 

Screen below comes from debugging facets produced locally. `CommitMetric` contain only 11 values, although `IcebergCommitMetrics` has over 20 fields. 

![Screenshot 2024-12-10 at 09 06 57](https://github.com/user-attachments/assets/46526c71-11c6-4417-a141-6b0f182e9cd3)